src/librustc/traits/error_reporting.rs

```diff
@@ -357,6 +383,9 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         let trait_ref = *trait_ref.skip_binder();
 
         let mut flags = vec![];
+        flags.push((sym::item_context,
+            Some(self.describe_enclosure(obligation.cause.body_id).to_owned())));
```

Comment:
...and use `if let Some(descr) = self.describe_enclosure<...>` here?

Reply:
I think that with that change you'd fix all of [the errors I'm seeing](https://dev.azure.com/rust-lang/rust/_build/results?buildId=12155).